nixos/modules/services/mail/maddy.nix

```diff
@@ -270,6 +328,20 @@ in {
 
   config = mkIf cfg.enable {
 
+    assertions = [{
+      assertion = cfg.tls.loader == "file" -> (
+        cfg.tls.certificates != null &&
+        length cfg.tls.certificates > 0 &&
+        all (attrs: attrs ? certPath && attrs ? keyPath) cfg.tls.certificates
```

Comment:
I'm not sure, but shouldn't the existence of these attributes be checked by the submodule? (Not 100% sure)

Reply:
I updated it and now you'll get the assertion error message with tls enabled if `certificates` is null or empty and if keyPath or certPath is missing in the attribute set. Otherwise the error message might be too cryptic